eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -490,4 +495,15 @@ class EclairImpl(appKit: Kit) extends Eclair with Logging {
     val pubKeyFromSignature = Crypto.recoverPublicKey(signature, signedBytes, recoveryId)
     VerifiedMessage(valid = true, pubKeyFromSignature)
   }
+
+  private def getChannelDescs(shortChannelIds: Set[ShortChannelId])(implicit timeout: Timeout): Future[Set[ChannelDesc]] = {
```

Comment:
If `shortChannelIds` is empty, you should immediately return an empty set instead of making a call to the `Router` for no reason.